### PR TITLE
add recipe to cook events

### PR DESCRIPTION
**Pull request by ExcessiveAmountsOfZombies**
for https://github.com/PaperMC/Paper/issues/4761
has paper comments now

### CODE REVIEW

**Comment by HexedHero** [comment]
You don't need paper comments on imports

File: Spigot-API-Patches/0237-add-recipe-to-cook-events.patch [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
++        this.recipe = recipe;
++    }
++    // Paper end
++
++    // Paper start
++    @Nullable
++    public Recipe getRecipe() {
++        return recipe;
+     }
++    // Paper end
```
Comment (Machine-Maker) [inline Spigot-API-Patches/0237-add-recipe-to-cook-events.patch:46]: The above Paper - start/end section can probably be combined with the one above it. Kinda redundant to have a `// Paper end` and `// Paper start` comment right after it.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by Chew** [comment]
You can't edit patch files directly like that, you'll need to edit locally and rebuild.

File: Spigot-API-Patches/0237-add-recipe-to-cook-events.patch [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
++        this(block, source, result, null); // Paper
++    }
++
++    // Paper start
++    public BlockCookEvent(@NotNull final Block block, @NotNull final ItemStack source, @NotNull final ItemStack result, @Nullable Recipe recipe) {
++    super(block);
++    this.source = source;
++    this.result = result;
++    this.cancelled = false;
++    this.recipe = recipe;
```
Comment (Machine-Maker) [inline Spigot-API-Patches/0237-add-recipe-to-cook-events.patch]: Shouldn't this be indented another level?

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by Machine-Maker** [comment]
@ExcessiveAmountsOfZombies Why was this closed?

**Comment by ExcessiveAmountsOfZombies** [comment]
@Machine-Maker I did something with the branch and screwed everything up and couldn't figure out how to fix it.

**Comment by Machine-Maker** [comment]
got a problem if I remake it? patch author will be you.

**Comment by ExcessiveAmountsOfZombies** [comment]
If you'd like to, sure. I think the problem was that I tried to update the patch numbers and ended up duplicating them.

**Pull request closed without merging**